### PR TITLE
Modernize freud binding, build system, and CI.

### PR DESCRIPTION
**Pull request by joaander**
## Description

* Replace Cython with nanobind.
* Replace scikit-build with scikit-build-core.
* Replace flake8 and black with ruff.
* Replace cppcheck with a modern version of clang-tidy.
* Support Python 3.13.

A big thanks to the large number of people who helped complete this refactoring!

## Motivation and Context

Make it easier for developers to improve freud.

## Checklist:
- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/freud/blob/main/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Freud Contributor Agreement**](https://github.com/glotzerlab/freud/blob/main/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/freud/blob/main/doc/source/reference/credits.rst) in the pull request source branch.
- [x] I have updated the [Change log](https://github.com/glotzerlab/freud/blob/main/ChangeLog.md).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -5,56 +5,64 @@ body:
...
     attributes:
-      label: System Platform
-      description: What platform are you using?
+      label: Platform
+      description: What platform are you using? Select all that apply.
+      multiple: true
       options:
-        - MacOS
+        - CPU
+        - GPU
```
Comment (bdice) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: Does freud have GPU support now?

File: .github/CODEOWNERS
Comment (bdice) [inline .github/CODEOWNERS:1]: How will code reviews be managed going forward?

**Review by bdice — APPROVED** [review]
I was pinged on this as a code reviewer, so I wanted to stop by just to congratulate the team on the immense amount of work that has gone in here. I had a couple small questions, but nothing blocking. I offer my greatest thanks for all the effort in maintaining a project that was a fond part of my time in the Glotzer lab.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (joaander) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: No, this is a copy and paste error.

File: .github/CODEOWNERS
Comment (joaander) [reply .github/CODEOWNERS:1]: The PR submitter will request a review, typically over Slack or in person.

**Comment by joaander** [comment]
> I was pinged on this as a code reviewer, so I wanted to stop by just to congratulate the team on the immense amount of work that has gone in here. I had a couple small questions, but nothing blocking. I offer my greatest thanks for all the effort in maintaining a project that was a fond part of my time in the Glotzer lab.

Thanks for looking at this. If I recall correctly, you helped a lot with the boost::python to Cython transition. The extensive unit testing added as part of that was a huge help here. The thorough tests caught many corner cases as we ported the internals over to nanobind.

**Comment by tommy-waltmann** [comment]
Commenting late, but congrats on seeing this to the finish line!

**Comment by joaander** [comment]
> Commenting late, but congrats on seeing this to the finish line!

Thanks for getting this project off to a great start, @tommy-waltmann. Many first year students in the group were able to follow the patterns you established and port modules themselves. Your `ManagedArray` implementation worked perfectly.
